src/test/java/nebula/test/SupportedGradleVersion.java

```diff
@@ -1,7 +1,7 @@
 package nebula.test;
 
 public enum SupportedGradleVersion {
-    MIN("8.11.1"), MAX("9.1.0");
+    MIN("8.12.1"), MAX("9.1.0");
```

Comment:
for some reason 8.11.1 was causing weird classloader issues specifically in GroovyDslSpockTest. 
I think maybe there was some kind of poisoned cache or something, but I couldn't figure out a way to fix it, but changing this worked